app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts

```diff
@@ -0,0 +1,1437 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Database} from '@nozbe/watermelondb';
+import {IntlShape} from 'react-intl';
+
+import {doAppCall} from '@actions/remote/apps';
+import {fetchChannelById, fetchChannelByName, searchChannels} from '@actions/remote/channel';
+import {fetchUsersByIds, fetchUsersByUsernames, searchUsers} from '@actions/remote/user';
+import {AppCallResponseTypes, AppCallTypes, AppFieldTypes, COMMAND_SUGGESTION_ERROR} from '@constants/apps';
+import DatabaseManager from '@database/manager';
+import {queryChannelById, queryChannelByName} from '@queries/servers/channel';
+import {queryCurrentTeamId} from '@queries/servers/system';
+import {queryUsersById, queryUsersByUsername} from '@queries/servers/user';
+import ChannelModel from '@typings/database/models/servers/channel';
+import UserModel from '@typings/database/models/servers/user';
+import {createCallRequest, filterEmptyOptions} from '@utils/apps';
+
+import {getChannelSuggestions, getUserSuggestions, inTextMentionSuggestions} from './mentions';
+/* eslint-disable max-lines */
+
+export enum ParseState {
+    Start = 'Start',
+    Command = 'Command',
+    EndCommand = 'EndCommand',
+    CommandSeparator = 'CommandSeparator',
+    StartParameter = 'StartParameter',
+    ParameterSeparator = 'ParameterSeparator',
+    Flag1 = 'Flag1',
+    Flag = 'Flag',
+    FlagValueSeparator = 'FlagValueSeparator',
+    StartValue = 'StartValue',
+    NonspaceValue = 'NonspaceValue',
+    QuotedValue = 'QuotedValue',
+    TickValue = 'TickValue',
+    EndValue = 'EndValue',
+    EndQuotedValue = 'EndQuotedValue',
+    EndTickedValue = 'EndTickedValue',
+    Error = 'Error',
+    Rest = 'Rest',
+}
+
+interface FormsCache {
+    getForm: (location: string, binding: AppBinding) => Promise<{form?: AppForm; error?: string} | undefined>;
+}
+
+interface Intl {
+    formatMessage(config: {id: string; defaultMessage: string}, values?: {[name: string]: any}): string;
+}
+
+// TODO: Implemnet app bindings
```

Comment:
Ticket to be created before merging the PR.

Reply:
This file is from the integrations team. This file is also shared with webapp. Therefore, any changes here may cause trouble for them, since they would have to do the same changes in webapp.

@catalintomai heads up that some changes here are happening (because of the new architecture) and that maybe your team will have to look into this for V2 beta.